--- qemu-5.2.0/accel/tcg/cpu-exec.c	2020-12-08 09:59:44.000000000 -0700
+++ qemu-5.2.0-patched/accel/tcg/cpu-exec.c	2021-03-24 16:52:45.421661003 -0600
@@ -41,6 +41,8 @@
 #include "sysemu/cpu-timers.h"
 #include "sysemu/replay.h"
 
+#include "../patches/afl-qemu-cpu-inl.h"
+
 /* -icount align implementation. */
 
 typedef struct SyncClocks {
@@ -152,6 +154,8 @@
     int tb_exit;
     uint8_t *tb_ptr = itb->tc.ptr;
 
+    AFL_QEMU_CPU_SNIPPET2;
+
     qemu_log_mask_and_addr(CPU_LOG_EXEC, itb->pc,
                            "Trace %d: %p ["
                            TARGET_FMT_lx "/" TARGET_FMT_lx "/%#x] %s\n",
@@ -412,6 +416,7 @@
     if (tb == NULL) {
         mmap_lock();
         tb = tb_gen_code(cpu, pc, cs_base, flags, cf_mask);
+        AFL_QEMU_CPU_SNIPPET1;
         mmap_unlock();
         /* We add the TB in the virtual pc hash table for the fast lookup */
         qatomic_set(&cpu->tb_jmp_cache[tb_jmp_cache_hash_func(pc)], tb);
