diff -Naur qemu-5.2.0/linux-user/elfload.c qemu-5.2.0-patched/linux-user/elfload.c
--- qemu-5.2.0/linux-user/elfload.c	2020-12-08 09:59:44.000000000 -0700
+++ qemu-5.2.0-patched/linux-user/elfload.c	2021-03-24 16:46:21.889123559 -0600
@@ -26,6 +26,8 @@
 
 #define ELF_OSABI   ELFOSABI_SYSV
 
+extern abi_ulong afl_entry_point, afl_start_code, afl_end_code;
+
 /* from personality.h */
 
 /*
@@ -2757,6 +2759,8 @@
     }
 #endif
 
+    if (!afl_entry_point) afl_entry_point = info->entry;
+
     for (i = 0; i < ehdr->e_phnum; i++) {
         struct elf_phdr *eppnt = phdr + i;
         if (eppnt->p_type == PT_LOAD) {
@@ -2805,9 +2809,11 @@
             if (elf_prot & PROT_EXEC) {
                 if (vaddr < info->start_code) {
                     info->start_code = vaddr;
+                    if (!afl_start_code) afl_start_code = vaddr;
                 }
                 if (vaddr_ef > info->end_code) {
                     info->end_code = vaddr_ef;
+                    if (!afl_end_code) afl_end_code = vaddr_ef;
                 }
             }
             if (elf_prot & PROT_WRITE) {
