--- qemu-2.10.0/util/memfd.c	2019-11-09 12:00:04.884881024 +0100
+++ qemu-2.10.0/util/memfd_patched.c	2019-11-09 12:00:23.372914761 +0100
@@ -37,7 +37,7 @@
 #include <sys/syscall.h>
 #include <asm/unistd.h>
 
-static int memfd_create(const char *name, unsigned int flags)
+int memfd_create(const char *name, unsigned int flags)
 {
 #ifdef __NR_memfd_create
     return syscall(__NR_memfd_create, name, flags);
